cyto_dl/image/transforms/maxpool_resize.py

```diff
@@ -0,0 +1,139 @@
+        (e.g., 32 -> (32, 32) for 2D or (32, 32, 32) for 3D).
+        If a sequence is provided, its length must match the number of spatial dimensions
+        (2 for images, 3 for volumes).
+    allow_missing_keys: bool, optional
+        If True, skips missing keys in the input data without raising an error.
+        Default is False.
+
+    Raises
+    ------
+    TypeError
+        If the input for a key is not a PyTorch tensor.
+    ValueError
+        If the input tensor has an unsupported number of dimensions, if `spatial_size`
+        has an invalid length, or if the output spatial size is invalid (e.g., larger
+        than the input size).
+    KeyError
+        If a key is missing in the input data and `allow_missing_keys` is False.
+ 
+    """
+
+    def __init__(
+        self,
+        keys: Union[str, Sequence[str]],
+        spatial_size: Union[Sequence[int], int],
+        allow_missing_keys: bool = False,
+    ):
+        super().__init__()
+        self.keys = keys if isinstance(keys, (list, ListConfig)) else [keys]
+        self.allow_missing_keys = allow_missing_keys
+
+        # Store raw spatial_size for later validation
+        self.spatial_size = spatial_size
+
+        # Validate spatial_size length in advance
+        if isinstance(spatial_size, (list, tuple, ListConfig)):
+            if len(spatial_size) not in (2, 3):
+                raise ValueError(
+                    f"spatial_size sequence must have length 2 or 3, got length {len(spatial_size)}"
+                )
+
+    def __call__(self, data):
+        d = dict(data)
+        for key in self.keys:
+            if key not in d:
+                if self.allow_missing_keys:
+                    continue
+                raise KeyError(f"Key '{key}' not found in input data.")
+
+            x = d[key]
+            if not isinstance(x, torch.Tensor):
+                raise TypeError(f"Input '{key}' must be a PyTorch tensor, got {type(x)}")
+
+            # Determine expected tensor dimensions and spatial size length
+            input_dims = x.dim()
+            if input_dims == 3:  # [C, H, W]
+                expected_spatial_dims = 2
+                x = x.unsqueeze(0)  # Add batch dimension: [1, C, H, W]
+                squeeze = True
+            elif input_dims == 4:  # [B, C, H, W]
+                expected_spatial_dims = 2
+                squeeze = False
+            elif input_dims == 5:  # [B, C, D, H, W]
+                expected_spatial_dims = 3
+                squeeze = False
+            else:
+                raise ValueError(
+                    f"Input '{key}' must be a 3D ([C, H, W]), 4D ([B, C, H, W]), "
+                    f"or 5D ([B, C, D, H, W]) tensor, got shape {x.shape}"
+                )
+
+            # Normalize spatial_size to match expected number of spatial dimensions
+            try:
+                spatial_size = ensure_tuple_rep(self.spatial_size, expected_spatial_dims)
```

Comment:
Adding an explicit spatial_dims argument here would let you find the spatial size once in the init